hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/StateContext.java

```diff
@@ -655,41 +659,45 @@ public void execute(ExecutorService service, long time, TimeUnit unit)
     // we called stop DatanodeStateMachine, this sets state to SHUTDOWN, and
     // there is a chance of getting task as null.
     if (task != null) {
```

Comment:
Let's change the if-statement to return instead of add a new indentation.
```java
@@ -654,7 +658,11 @@ public void execute(ExecutorService service, long time, TimeUnit unit)
     // Adding not null check, in a case where datanode is still starting up, but
     // we called stop DatanodeStateMachine, this sets state to SHUTDOWN, and
     // there is a chance of getting task as null.
-    if (task != null) {
+    if (task == null) {
+      return;
+    }
+
+    try {
       if (this.isEntering()) {
         task.onEnter();
       }
@@ -691,6 +699,8 @@ public void execute(ExecutorService service, long time, TimeUnit unit)
         // that we can terminate the datanode.
         setShutdownOnError();
       }
+    } finally {
+      task.clear();
     }
   }
```

Reply:
I'll update later.